llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -6718,6 +6718,13 @@ SIInstrInfo::legalizeOperands(MachineInstr &MI,
     if (SRsrc && !RI.isSGPRClass(MRI.getRegClass(SRsrc->getReg())))
       CreatedBB = loadMBUFScalarOperandsFromVGPR(*this, MI, {SRsrc}, MDT);
 
+    if (isMUBUF(MI) || isMTBUF(MI)) {
```

Comment:
We shouldn't really be restricting the opcodes for this. This should just be the behavior for any scalar required operand, including inline asm. Why wouldn't this cover images? The calling convention check above is also nonsense